pkg/volume/csi/csi_attacher.go

```diff
@@ -293,11 +276,12 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 
 	// Store volume metadata for UnmountDevice. Keep it around even if the
 	// driver does not support NodeStage, UnmountDevice still needs it.
-	if err = os.MkdirAll(deviceMountPath, 0750); err != nil && !corruptedDir {
-		return errors.New(log("attacher.MountDevice failed to create dir %#v:  %v", deviceMountPath, err))
+	parentDir := filepath.Dir(deviceMountPath)
+	if err = os.MkdirAll(parentDir, 0750); err != nil {
+		return errors.New(log("attacher.MountDevice failed to create dir %#v:  %v", parentDir, err))
 	}
-	klog.V(4).Info(log("created target path successfully [%s]", deviceMountPath))
-	dataDir := filepath.Dir(deviceMountPath)
+	klog.V(4).Info(log("created target path successfully [%s]", parentDir))
+	dataDir := parentDir
```

Comment:
@andyzhangx @msau42 I think we have to create staging path at least, to be spec complaint.